### DISCUSSION TITLE
Issue #3918

### DISCUSSION THREAD

**Discussion by doncicuto** [zitadel/zitadel#5115] [Contributors]
Hi @hifabienne and @juergrinaldi, if you don't mind I'd like to discuss my proposed solution for #3918 before wasting your time reviewing the PR :smile:

**Comment by doncicuto** [comment]
### Design issue 1

I've chosen to use npm package [flag-icons](https://www.npmjs.com/package/flag-icons) (MIT license) that provides convenient SVG flags and CSS to ease development. Due to that I'd suggest removing PNG flags located at assets as I think that aren't used in Zitadel project.

**Comment by doncicuto** [comment]
### Design issue 2

When a new user is created, a default country for phone number has to be set (show flag). 

![Captura desde 2023-01-28 11-04-06](https://user-images.githubusercontent.com/30386061/215275542-4d94bcd5-3569-4728-ac6c-835910ec51f8.png)


I could have used a GeoIP database/service to get the country from IP address but I wanted to reduce dependencies and as Zitadel may be used in a private environment I discarded it. Also I could have tried to use browser's location API but I thought that it would be cumbersome for users to ask for permission anytime visiting Zitadel. 

I finally thought that I may use timezone to provide a guess about user's country home. Although timezone is not the best way to assign a country as there are timezones used by several countries, I think it's better than nothing. If timezone doesn't work, Switzerland will be the default country.

NPM package [countries-and-timezones](https://www.npmjs.com/package/countries-and-timezones) (MIT license) helps me to assign the probable country according to the timezone.

**Reply by juergrinaldi** [reply]
Since selecting a default is a bit a tricky one here, would it be an option to setup an (optional) default from the org admin? (i.e. a e-shop owner in Ireland can set IRE +353 as the default for their new users). @hifabienne @peintnermax would that be feasable?

I don't think the timezone one is feasable (nice idea though!) as some timezones have A LOT of countries in (i.e. GMT +2 has around 30 or so) and that just seems pretty random imo. in that case, I'd rather go with a generic default. (either Switzerland, because ZITADEL is Switzerland based, or USA as it is the first entry and probably also defaults for bigger population).

**Reply by doncicuto** [reply]
I agree @juergrinaldi, the timezone is not enough, so I'll remove that code from my proposal. For now I'll set Switzerland as the default country (flag shown) and if a country is added to the organization I may use that instead.

**Comment by doncicuto** [comment]
### Design issue 3

When a user clicks the flags dropdown menu I thought it would show country names according to the user's account selected language, so I've added [i18n-iso-countries](https://www.npmjs.com/package/i18n-iso-countries) from NPM (MIT license) that gives me country names in Zitadel's supported languages.

![Captura desde 2023-01-28 16-42-36](https://user-images.githubusercontent.com/30386061/215275685-b858b54e-400c-4434-9038-35eb6c841446.png)

**Reply by juergrinaldi** [reply]
That's probably already setup but just a hint that by clicking on the dropdown/flag the user should be enabled to type the country's name or the number for easier search of their country.

**Reply by doncicuto** [reply]
Hi @juergrinaldi, I could try to add a css box under the dropdown and phone input explaining "You can add a phone number using 00 or + followed by the country calling code or choose the flag from the dropdown to pick the country calling code, then type the phone number" or maybe a tooltip. Would that be good enough to offer a hint for users to understand the control?

The dropdown menu hasn't a search option (you can't type the country name to go to the right flag). Google Contact doesn't have one. Do you need a search option for the dropdown?

**Reply by doncicuto** [reply]
Hi @juergrinaldi, I have finally added a tooltip explaining how you can enter the phone either typing the country calling code or using the dropdown.

![image](https://user-images.githubusercontent.com/30386061/215562240-bd42f4ee-40cc-41ac-83a0-b1806c977995.png)

**Reply by juergrinaldi** [reply]
ah sorry, I probably wrote that a bit missleading. I just wanted to express that the user should be able to use either + or 00, not that the user have to have a hint on that. My bad!
But your changes makes it even clearer this way. :)

And no, I didn't want a specific search box but just to be able to type when you're in the dropdown. (as Google Contacts allows as well).

Sorry for the confusion.

**Comment by doncicuto** [comment]
### Design issue 4

I think the flag should be shown next to the formatted phone number in user's detail.

![Captura desde 2023-01-28 11-03-24](https://user-images.githubusercontent.com/30386061/215275724-4237a53a-a5fb-46b8-81c8-87ee547bb060.png)

**Comment by doncicuto** [comment]
And finally, this a video showing my proposal. Once you feel this proposal is completed I'll send my PR. 

https://user-images.githubusercontent.com/30386061/215276968-0e7db07f-f493-46c5-9cc1-97bd96fd3a48.mp4

Thanks!

**Comment by juergrinaldi** [comment]
Looks awesome @doncicuto . I really dig the changes you did there, Miguel. 👍

**Reply by doncicuto** [reply]
Glad you like it, I'll do some mods to my current code adding any your suggestions. I'm having fun browsing good-first-issues for Zitadel.

**Reply by juergrinaldi** [reply]
Glad you're having a blast. :)
We always love when our community helps to improve ZITADEL with us. Feel free to take on other issues or submit other suggestions and ideas. Always welcome.

**Reply by doncicuto** [reply]
Awesome, thanks @juergrinaldi for your help. I've created a PR with the code at https://github.com/zitadel/zitadel/pull/5139, looking forward for your feedback or improvements required.

**Comment by doncicuto** [comment]
This discussion can be closed. Thanks!
